test/spec/unit/core/storageManager_spec.js

```diff
@@ -42,5 +42,22 @@ describe('storage manager', function() {
     storage.setCookie('foo1', 'baz1');
     expect(deviceAccessSpy.calledOnce).to.equal(true);
     deviceAccessSpy.restore();
+  });
+
+  it('should not throw if the localstorage is not accessible when setting/getting/removing from localstorage', function() {
+    const coreStorage = getStorageManager();
+
+    const localstorageStub = sinon.stub(global.window, 'localStorage').get(() => { throw Error });
+    const errorLogSpy = sinon.spy(utils, 'logError');
+
+    coreStorage.setDataInLocalStorage('key', 'value');
+    const val = coreStorage.getDataFromLocalStorage('key');
+    coreStorage.removeDataFromLocalStorage('key');
+
+    expect(val).to.be.null;
+    sinon.assert.calledThrice(errorLogSpy);
+
+    localstorageStub.restore();
+    errorLogSpy.restore();
```

Comment:
Can you move the stubs (and the subsequent restores) into a before/after function surrounding this test?  This will help ensure these objects are properly handled in case the test fails.  If you don't want to have these stubs effect the other tests, please feel free to create another describe block to nest it altogether.

Reply:
Ok 👍 

> Btw, in this case what is "third party context"? Is that just JS loaded from 3p domain? 
> > access window.localstorage in a third-party context
Indeed, 3rd-party context refers to scripts run inside an iframe whose src domain is not the 1p domain for instance.